File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s081988820_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func readAllInts() []int {
	scanner := bufio.NewScanner(os.Stdin)
	var nums []int
	for scanner.Scan() {
		line := scanner.Text()
		fields := strings.Fields(line)
		for _, f := range fields {
			num, _ := strconv.Atoi(f)
			nums = append(nums, num)
		}
	}
	if err := scanner.Err(); err != nil {
		panic(err)
	}
	return nums
}

func main() {
	nums := readAllInts()
	H := nums[0]
	W := nums[1]
	index := 2

	c := make([][]int, 10)
	for i := range c {
		c[i] = make([]int, 10)
		for j := 0; j < 10; j++ {
			c[i][j] = nums[index]
			index++
		}
	}

	min := make([]int, 10)
	for i := 0; i < 10; i++ {
		min[i] = c[i][1]
	}

	for tc := 0; tc < 10; tc++ {
		for i := 0; i < 10; i++ {
			for j := 0; j < 10; j++ {
				if c[i][j]+min[j] < min[i] {
					min[i] = c[i][j] + min[j]
				}
			}
		}
	}

	ans := 0
	for ; index < len(nums); {
		A := nums[index]
		index++
		if A >= 0 {
			ans += min[A]
		}
	}
	fmt.Println(ans)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C++/s081988820_comments_removed.cpp
#include <iostream>
using namespace std;

class Main {
public:
    void solve() {
        int H, W;
        cin >> H >> W;
        int c[10][10];
        for (int i = 0; i < 10; ++i) {
            for (int j = 0; j < 10; ++j) {
                cin >> c[i][j];
            }
        }
        int min[10];
        for (int i = 0; i < 10; ++i) {
            min[i] = c[i][1];
        }
        for (int tc = 0; tc < 10; ++tc) {
            for (int i = 0; i < 10; ++i) {
                for (int j = 0; j < 10; ++j) {
                    if (min[i] > c[i][j] + min[j]) {
                        min[i] = c[i][j] + min[j];
                    }
                }
            }
        }
        int ans = 0;
        for (int h = 1; h <= H; ++h) {
            for (int w = 1; w <= W; ++w) {
                int A;
                cin >> A;
                if (A >= 0) {
                    ans += min[A];
                }
            }
        }
        cout << ans << endl;
    }
};

int main() {
    Main m;
    m.solve();
    return 0;
}
// End of Code
